From f5f130c951c98143ed4cccdebf6c0faf59903de7 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Wed, 7 Mar 2018 22:57:50 +0200
Subject: [PATCH 1046/5242] MLK-12757-3: ARM: dts: imx6ull-ddr3-arm2-emmc.dts:
 add eMMC support

commit  b67dd1b5a7891fd14acb336492501f85a7a91013 from
https://source.codeaurora.org/external/imx/linux-imx.git

For imx6ull-ddr3-arm2 board, eMMC and SD1 slot share usdhc1, so
this patch add another dts file imx6ull-ddr3-arm2-emmc.dts to
support eMMC.

eMMC data4~data7 share the same I/O domain with sd2, so this patch
only enable eMMC 4bit mode.

Due to the eMMC on imx6ull-ddr3-arm2 board support HS200 mode, and
need the VCCQ to be 1.8v, this patch keep usdhc vselect, let usdhc
to change the I/O voltage to 1.8v automatically. Otherwise, another
rework needed: remove R95, add R94.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                         |    1 +
 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-emmc.dts |   20 ++++++++++++++++++++
 2 files changed, 21 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-emmc.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index af6de85..ced452a 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -602,6 +602,7 @@ dtb-$(CONFIG_SOC_IMX6ULL) += \
 	imx6ull-14x14-ddr3-arm2-adc.dtb \
 	imx6ull-14x14-ddr3-arm2-cs42888.dtb \
 	imx6ull-14x14-ddr3-arm2-ecspi.dtb \
+	imx6ull-14x14-ddr3-arm2-emmc.dtb \
 	imx6ull-14x14-ddr3-arm2-flexcan2.dtb \
 	imx6ull-14x14-ddr3-arm2-lcdif.dtb \
 	imx6ull-14x14-ddr3-arm2-qspi.dtb \
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-emmc.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-emmc.dts
new file mode 100644
index 0000000..7b24466
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-emmc.dts
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6ull-14x14-ddr3-arm2.dts"
+
+&usdhc1 {
+	assigned-clocks = <&clks IMX6UL_CLK_USDHC1_SEL>, <&clks IMX6UL_CLK_USDHC1>;
+	assigned-clock-parents = <&clks IMX6UL_CLK_PLL2_PFD0>;
+	assigned-clock-rates = <0>, <176000000>;
+	cd-gpios = <>;
+	wp-gpios = <>;
+	vmmc-supply = <>;
+	non-removable;
+	status = "okay";
+};
-- 
1.7.9.5

